### ISSUE TITLE
tf.image.rot90 has inconsistent results in different TensorFlow versions.

### ISSUE THREAD

**Issue by triumph-wangyuyang** [tensorflow/tensorflow#58702]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

TF 2.4, TF 2.10

### Custom Code

Yes

### OS Platform and Distribution

windows 11

### Python version

3.8.15

### CUDA/cuDNN version

CUDA: 11.3    cuDNN 8.6.0

### GPU model and memory

RTX3060

### Current Behaviour?

```shell
Comparing TensorFlow2.4 and TensorFlow2.10, it is found that the description of tf.image.rot90 is consistent, but the results of running TensorFlow2.4 and TensorFlow2.10 are inconsistent. It will fail in the lower version, but will succeed in the higher version. Judging from the exception thrown in the lower version, the data type is not supported, and the higher version supports this type, but this type of restriction is not proposed in the official document, which will cause misunderstanding.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

    with tf.device('/CPU'):
        image = tf.saturate_cast(tf.random.uniform([2, 2, 1], minval=0, maxval=257, dtype=tf.int64), dtype=tf.uint64)
        out = tf.image.rot90(image)
        print(out)
```

### Relevant log output

```shell
Results running on TensorFlow 2.4:  
tensorflow.python.framework.errors_impl.InvalidArgumentError: Value for attr 'T' of uint64 is not in the list of allowed values: uint8, int8, uint16, int16, int32, int64, bool, bfloat16, half, float, double, complex64, complex128, string; NodeDef: {{node ReverseV2}}; Op<name=ReverseV2; signature=tensor:T, axis:Tidx -> output:T; attr=Tidx:type,default=DT_INT32,allowed=[DT_INT32, DT_INT64]; attr=T:type,allowed=[DT_UINT8, DT_INT8, DT_UINT16, DT_INT16, DT_INT32, DT_INT64, DT_BOOL, DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE, DT_COMPLEX64, DT_COMPLEX128, DT_STRING]> [Op:ReverseV2]


Results running on TensorFlow 2.10:  
tf.Tensor(
[[[72]
  [70]]

 [[35]
  [12]]], shape=(2, 2, 1), dtype=uint64)
```
</details>

**Comment by triumph-wangyuyang** [comment]
tf.image.stateless_random_flip_up_down will also have inconsistent results in different TensorFlow versions (such as TensorFlow2.4 and TensorFlow2.10).

code:

    import tensorflow as tf
    image = tf.saturate_cast(tf.random.uniform([2, 2, 1], minval=0, maxval=257, dtype=tf.int64), dtype=tf.uint64)
    seed = [2, 3]
    out = tf.image.stateless_random_flip_up_down(image, seed)

**Comment by tilakrayal** [comment]
@triumph-wangyuyang,
In tf v2.4, it was executed with the error and whereas in the latest tf version, the dtype was supported and able to execute without any issue/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/b1100753e80bc279e1770be1d36e3b58/untitled781.ipynb). Thank you!

**Comment by tilakrayal** [comment]
@triumph-wangyuyang,
From the [gist](https://colab.research.google.com/gist/tilakrayal/b1100753e80bc279e1770be1d36e3b58/untitled781.ipynb) it was clear that the issue was resolved in the latest tf v2.11 and please feel free to move this to closed status. Thank you!

**Comment by triumph-wangyuyang** [comment]
> @triumph-wangyuyang, From the [gist](https://colab.research.google.com/gist/tilakrayal/b1100753e80bc279e1770be1d36e3b58/untitled781.ipynb) it was clear that the issue was resolved in the latest tf v2.11 and please feel free to move this to closed status. Thank you!

thank you for your reply
